content/blog/2023/07/02/2023-07-02-unreleased-production-changes-probe.adoc

```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title: Unreleased Production Changes Probe
+tags:
+- gsoc
+- gsoc2023
+- healthscore
+- probes
+- plugin
+authors:
+- jagruti
+opengraph:
+  image: /images/gsoc/2023/jagruti/blog_20230702.png
+sig:
+- gsoc
+---
+
+== Introduction
+Unreleased production changes probe looks for unreleased commits in the production files.
+
+If an unreleased file is found in the production path, the probe returns a failure message. If no unreleased file is found, it returns a success message.
+
+The probe commenced during the GSoC acceptance waiting period.
+
+This is my first probe that went live as a GSoC contributor.
+
+== Importance of the probe
+Finding an unreleased production file can mean one or more of the following:
+
+* The plugin may have undelivered bug fixes.
+* The plugin may have stale features.
+* The maintainer of the plugin maybe inactive.
+
+== Probe details
+The probe aims to identify unreleased production commits in `src/main` directory and `pom.xml` file.
+
+== Challenges
+Implementing https://www.eclipse.org/jgit/[JGit], a Java library that implements Git version control system was one of the biggest challenges I faced when working on this probe. Also, I was not familiar with Git hierarchy. I struggled to find the parent of a child commit using JGit.
+
+In the end, my mentors suggested a pair programming session. We rewrote the probe together. I am thankful to their kindness and patience.
+
+== Outcome
+In the case when unreleased production files are found, the probe returns a list with their names and a failure message is returned. In the case when no unreleased production file is found, the probe returns a success message.
+
+== Learning
+* Learned to figure out unit test cases and implemented them.
+* Implemented JGit Java library.
+* Understood Git commit hierarchy.
```

Comment:
Same as before, telling the story rather than listing the items could be better.